exec/java-exec/src/main/java/org/apache/drill/exec/store/sys/store/EphemeralPersistentStore.java

```diff
@@ -93,6 +107,14 @@ public void put(final String key, final V value, final DataChangeVersion dataCha
         throw new VersionMismatchException("Version mismatch detected", dataChangeVersion.getVersion());
       }
       store.put(key, value);
+      currentSize.incrementAndGet();
+
+      if (currentSize.get() > maxCapacity) {
+        //Pop Out Oldest
+        ((ConcurrentSkipListMap<String, V>) store).pollLastEntry();
```

Comment:
The natural ordering of the profiles actually has the 'smallest' key being the latest. Hence, we pop out the oldest entry by getting the 'largest'/last entry